### PR TITLE
[MI-3273] Review fixes on ServiceNow PR #182 (Add create incident modal)

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
Link: https://github.com/mattermost/mattermost-plugin-servicenow/pull/182

### CODE REVIEW

File: webapp/src/containers/addOrEditSubscriptions/subComponents/index.tsx [hunk 1]
```diff
@@ -88,13 +88,13 @@ const AddOrEditSubscription = ({open, close, subscriptionData}: AddOrEditSubscri
...
         const {isLoading, isSuccess, isError, data, error: apiErr} = getApiState(Constants.pluginApiServiceConfigs.createSubscription.apiServiceName, createSubscriptionPayload as CreateSubscriptionPayload);
-        return {isLoading, isSuccess, isError, data: data as RecordData, error: (apiErr as FetchBaseQueryError)?.data as APIError | undefined};
+        return {isLoading, isSuccess, isError, data: data as RecordData, error: apiErr};
     };
 
     // Get edit subscription state
     const getEditSubscriptionState = () => {
         const {isLoading, isSuccess, isError, data, error: apiErr} = getApiState(Constants.pluginApiServiceConfigs.editSubscription.apiServiceName, editSubscriptionPayload as EditSubscriptionPayload);
-        return {isLoading, isSuccess, isError, data: data as RecordData, error: (apiErr as FetchBaseQueryError)?.data as APIError | undefined};
+        return {isLoading, isSuccess, isError, data: data as RecordData, error: apiErr};
```
[the remark this answers is not in the document]
Comment (raghavaggarwal2308) [reply webapp/src/containers/addOrEditSubscriptions/subComponents/index.tsx:97]: We are getting a variable named `error` in props

File: webapp/src/containers/addOrEditSubscriptions/subComponents/index.tsx [hunk 2]
```diff
@@ -88,13 +88,13 @@ const AddOrEditSubscription = ({open, close, subscriptionData}: AddOrEditSubscri
     // Get create subscription state
     const getCreateSubscriptionState = () => {
         const {isLoading, isSuccess, isError, data, error: apiErr} = getApiState(Constants.pluginApiServiceConfigs.createSubscription.apiServiceName, createSubscriptionPayload as CreateSubscriptionPayload);
-        return {isLoading, isSuccess, isError, data: data as RecordData, error: (apiErr as FetchBaseQueryError)?.data as APIError | undefined};
+        return {isLoading, isSuccess, isError, data: data as RecordData, error: apiErr};
```
[the remark this answers is not in the document]
Comment (raghavaggarwal2308) [reply webapp/src/containers/addOrEditSubscriptions/subComponents/index.tsx:91]: same

**Review by ayusht2810 — APPROVED** [review]

File: server/plugin/util.go [hunk 3]
```diff
@@ -309,10 +309,9 @@ func decodeKey(key string) (string, error) {
 }
 
 func (p *Plugin) HasChannelPermissions(userID, channelID string) (int, error) {
-	// Check if a user is a part of the channel
-	if _, channelErr := p.API.GetChannelMember(channelID, userID); channelErr != nil {
-		p.API.LogDebug(constants.ErrorChannelPermissionsForUser, "Error", channelErr.Error())
-		return channelErr.StatusCode, fmt.Errorf(constants.ErrorInsufficientPermissions)
+	if !p.API.HasPermissionToChannel(userID, channelID, model.PermissionCreatePost) {
+		p.API.LogDebug(constants.ErrorChannelPermissionsForUser)
```
Comment (ayusht2810) [inline server/plugin/util.go:313]: I think we should add userID and channelID in these logs.

File: webapp/src/containers/Rhs/index.tsx [hunk 4]
```diff
@@ -260,7 +260,7 @@ const Rhs = (): JSX.Element => {
                                 subTitle={isCurrentUserSysAdmin ? Constants.SubscriptionsConfigErrorSubtitleForAdmin : Constants.SubscriptionsConfigErrorSubtitleForUser}
                                 buttonConfig={isCurrentUserSysAdmin ? ({
                                     text: 'Download update set',
-                                    link: Utils.getBaseUrls().publicFilesUrl + UPLOAD_SET_FILENAME,
+                                    link: Utils.getBaseUrls(SiteURL as string).publicFilesUrl + UPLOAD_SET_FILENAME,
```
Comment (ayusht2810) [inline webapp/src/containers/Rhs/index.tsx:263]: can we typecast it once above to avoid writing this again?

**Review by ayusht2810 — APPROVED** [review]

File: webapp/src/containers/createIncident/subscribeToNewIncident.tsx [hunk 5]
```diff
@@ -66,7 +66,7 @@ const SubscribeNewIncident = ({
                 setShowResultPanel(true);
             }
         }
-    }, [getSubscriptionState().isError, getSubscriptionState().isSuccess, getSubscriptionState().isLoading]);
+    }, [subscriptionPayload, isLoading, isError, isSuccess]);
```
Comment (manojmalik20) [inline webapp/src/containers/createIncident/subscribeToNewIncident.tsx:69]: `subscriptionPayload` wasn't present here before. Is there really a need for this?

File: webapp/src/services/index.ts [hunk 6]
```diff
@@ -1,13 +1,40 @@
+// eslint-disable-next-line import/no-unresolved
```
Comment (manojmalik20) [inline webapp/src/services/index.ts:1]: What's the need for this comment?

File: webapp/src/containers/createIncident/subscribeToNewIncident.tsx (hunk 5 above)
Comment (raghavaggarwal2308) [reply webapp/src/containers/createIncident/subscribeToNewIncident.tsx:69]: https://github.com/mattermost/mattermost-plugin-servicenow/pull/182#discussion_r1253650334

File: webapp/src/services/index.ts (hunk 6 above)
Comment (raghavaggarwal2308) [reply webapp/src/services/index.ts:1]: The line below was failing on CI. Took this from mm-ui-lib.

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**